lib/elixir/lib/exception.ex

```diff
@@ -1492,7 +1492,11 @@ defmodule MatchError do
 
   @impl true
   def message(exception) do
-    "no match of right hand side value: #{inspect(exception.term)}"
+    """
+    no match of right hand side value:
+
+    #{inspect(exception.term, pretty: true, limit: :infinity)}
```

Comment:
We can pretty print but we should not set it to infinity. Given this goes to production logs, it can have a pretty negative impact.